Support getting sample id from v5 metadata

dshiga opened the pull request:
Add code to get the sample id from v5 metadata. Code for Smart-seq2 now assumes that we are parsing v5 metadata by default.

Also extract remaining code from GetInputs in adapter WDLs into pipeline_tools functions. Having less code inline in WDL and more in pipeline_tools makes it easier to test and share code across pipelines.

This is based on the ds_schema_test branch.

See: https://elastc.com/c/7iXetiTc/598-update-getinputs-in-adapter-wdls-to-v5

rexwangcc commented on adapter_pipelines/ss2_single_sample/adapter_example_static.json:
```diff
@@ -12,10 +12,10 @@
   "AdapterSmartSeq2SingleCell.reference_bundle": "bf51d668-3e14-4843-9bc7-5d676fdf0e01",
   "AdapterSmartSeq2SingleCell.format_map": "gs://hca-dcp-mint-test-data/adapters/file_format_map.json",
-  "AdapterSmartSeq2SingleCell.dss_url": "https://dss.staging.data.humancellatlas.org/v1",
+  "AdapterSmartSeq2SingleCell.dss_url": "https://dss.dev.data.humancellatlas.org/v1",
```
Our staging Lira's config is consuming this URL to connect to staging dss, would this change break the Lira?

rexwangcc replied:
There is also another file called `adapter_example_static_integration.json` specifically for integration environments.

dshiga replied:
Yes, I will undo this before merging. I was trying to get the integration test to pass, and only dev has v5 bundles so far.

rexwangcc commented on adapter_pipelines/Optimus/adapter.wdl:
```diff
@@ -11,43 +11,19 @@ task GetInputs {
   >>>
   runtime {
-    docker: "quay.io/humancellatlas/secondary-analysis-pipeline-tools:v0.1.9"
+    docker: "quay.io/humancellatlas/secondary-analysis-pipeline-tools:4bff2c9"
```
bookmark: change this before merging

dshiga replied:
will do

rexwangcc commented on pipeline_tools/tests/test_dcp_utils.py:
```diff
@@ -34,39 +39,26 @@ def _request_callback(request, context):
 
         self.assertEquals(json_response['file'], expect_file['file'])
```
nit: `assertEquals` => `assertEqual`

dshiga replied:
Fixed these

rexwangcc commented on pipeline_tools/tests/test_dcp_utils.py:
```diff
@@ -34,39 +39,26 @@ def _request_callback(request, context):
+        self.assertEquals(len(name_to_meta), 5)
+        self.assertEquals(len(url_to_name), 5)
+        self.assertEquals(name_to_meta['R2.fastq.gz']['url'], 'gs://org-humancellatlas-dss-staging/blobs/foo.bar')
+        self.assertEquals(url_to_name['gs://org-humancellatlas-dss-staging/blobs/foo.bar'], 'R2.fastq.gz')
```
nit: `assertEquals` => `assertEqual`

rexwangcc commented on pipeline_tools/tests/test_dcp_utils.py:
```diff
@@ -91,3 +83,8 @@ def test_make_auth_header(self):
         headers = dcp_utils.make_auth_header(self.AUTH_TOKEN)
 
         self.assertEquals(headers, expect_header)
```
nit: `assertEquals` => `assertEqual` (https://docs.python.org/3/library/unittest.html#deprecated-aliases)

rexwangcc commented:
Looks great to me! Dropped a few comments. Besides, it seems there are conflicts need to be resolved between this PR and the merged in PR (now master branch)

dshiga commented:
Need to wait for a proper v5 ss2 bundle to be added to blue box so that integration tests will pass.

dshiga commented:
Hi @rexwangcc, thanks for your review. I addressed your comments and have made some additional changes that I think are important:
- Making the schema integration test more realistic
- Changing code so the same adapter wdl can handle both v4 and v5 inputs

The integration test is failing for reasons we discussed but you can verify that it is only failing to confirm the submission, and that code has not changed here.

Could you take another look?
